src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageAction.java

```diff
@@ -40,9 +41,13 @@ protected ActionResult execute() throws EntityDoesNotExistException {
 
         InstructorFeedbackResultsPageData data = new InstructorFeedbackResultsPageData(account, sessionToken);
         String selectedSection = getRequestParamValue(Const.ParamsNames.FEEDBACK_RESULTS_GROUPBYSECTION);
+        SectionDetail selectedSectionDetail = null;
+        String sectionDetailValue = getRequestParamValue(Const.ParamsNames.FEEDBACK_RESULTS_GROUPBYSECTIONDETAIL);
 
         if (selectedSection == null) {
             selectedSection = ALL_SECTION_OPTION;
+        } else if (sectionDetailValue != null && !sectionDetailValue.isEmpty()) {
```

Comment:
Under what situation will sectionDetailsValue be `null` or empty?

Reply:
sectionDetailsValue is null when the `selectedSection` is all. When a session result is shown without the option to choose a filter section, the sectionDetailsValue is empty.

Reply:
Maybe you want to use a special value in the enum instead of making it `null`.

Reply:
In this case, do you suggest a "DEFAULT" value in enum then?

Reply:
If `ALL_SECTION_OPTION` is selected and you decide to ignore the value, I believe `NOT_APPLICABLE` is more appropriate.